### PR TITLE
Set up models, and migrations

### PR DESCRIPTION
**Pull request by Cort98**
added piece model, edited games model to include player ids
hooked up pieces to games, and games to user
add piece resource under games

### CODE REVIEW

File: db/migrate/20160728023902_create_pieces.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+  def change
+    create_table :pieces do |t|
+      t.string   :type
+      t.integer  :game_id
+      t.string   :color
+      t.integer  :column_coordinate
+      t.integer  :row_coordinate
+
+      t.timestamps
+    end
```
Comment (mscoutermarsh) [inline db/migrate/20160728023902_create_pieces.rb]: Looks good!

Next thing to do: You'll want to add indexes on the associations.

Take a look at this article: https://tomafro.net/2009/08/using-indexes-in-rails-index-your-associations.

It talks about MySQL, but the concept is the same. Let me know if any questions :). You can add them on to this PR.

File: db/migrate/20160728193153_add_index_to_pieces.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class AddIndexToPieces < ActiveRecord::Migration
+  def change
+  end
+  add_index :pieces, :game_id
```
Comment (mscoutermarsh) [inline db/migrate/20160728193153_add_index_to_pieces.rb]: You'll want this line to be inside

```Ruby
def change
  add_index :pieces, :game_id
end
```

The `change` method makes it easy for that migration to be run/rolledback. If someone runs db:rollback on that migration. `change` will do the opposite of adding the index (remove it).

File: test/models/piece_test.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require 'test_helper'
+
+class PieceTest < ActiveSupport::TestCase
+  # test "the truth" do
+  #   assert true
+  # end
+end
```
Comment (mscoutermarsh) [inline test/models/piece_test.rb]: Can you delete this file, since we're switching to rspec. We won't need it.

**Comment by tohyung85** [comment]
Hey guys, i basically added indexes to the white player and black player ids in the games table

**Comment by tohyung85** [comment]
Set up associations between User and Games
    
    In Games.rb
    - Create belongs_to relationship with white_player
    - Create belongs_to relationship with black_player
    
    In User.rb
    - Create has_many relationship with white_player_games
    - Create has_many relationship with black_player_games
    - Method to return all games user is involved in. 
(has_many :games won't work because we have 2keys in our games table)

File: app/models/user.rb [hunk 4]
```diff
@@ -3,4 +3,10 @@ class User < ActiveRecord::Base
   # :confirmable, :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
+  has_many :white_player_games, :class_name => 'Game', :foreign_key => 'white_player_id'
+  has_many :black_player_games, :class_name => 'Game', :foreign_key => 'black_player_id'
+
+  def games
+    Game.where("white_player_id = ? OR black_player_id = ?", self.id, self.id)
+  end
```
Comment (mscoutermarsh) [inline app/models/user.rb]: nicely done!

File: db/migrate/20160728193153_add_index_to_pieces.rb (hunk 2 above)
Comment (mscoutermarsh) [inline db/migrate/20160728193153_add_index_to_pieces.rb]: If you can fix this 1 thing ^. Then this PR is good to merge (go ahead and merge it yourself) 😄

**Pull request merged**